src/components/ModalLedgerSignToken.js

```diff
@@ -0,0 +1,264 @@
+   * @param {Object} arg Data returned from the send token data call
+   */
+  const handleSignature = (event, arg) => {
+    console.log('handleSignature', arg);
+    setWaitingLedger(false);
+    if (arg.success) {
+      // save token signature on storage
+      tokens.addTokenSignature(token.uid, arg.data.toString("hex"));
+      setOk(true);
+    } else {
+      switch (arg.error.status) {
+        // user deny
+        case 0x6985: 
+          setError('user_deny');
+          break;
+        case 0xb00a: 
+          setError('invalid_token');
+          break;
+        default:
+          setError('unknown_error');
+      }
+    }
+  }
+
+  /**
+   * Handle the response of a send token data call to Ledger.
+   * Will list all tokens that failed verification, if all pass then start sign tx
+   *
+   * @param {IpcRendererEvent} event May be used to reply to the event
+   * @param {Object} arg Data returned from the send token data call
+   */
+  const handleVerify = (event, arg) => {
+    console.log('handleVerify', arg);
+    setWaitingLedger(false);
+    if (arg.success) {
+      setOk(true);
+    } else {
+      // Invalid signature, we should start the signing process
+      canSend(true);
+    }
+  }
+
+  /**
+   * Effects
+   */
+
+  // useEffect second parameter [] ensures this runs only once
+  useEffect(() => {
+    $(`#${modalId}`).on('hide.bs.modal', (e) => {
+      // clean state when closing
+      initialState();
+      // remove ledger listeners when closing
+      if (IPC_RENDERER) {
+        IPC_RENDERER.removeAllListeners("ledger:tokenSignature");
+        IPC_RENDERER.removeAllListeners("ledger:tokenSignatureValid");
+      }
+    });
+
+    $(`#${modalId}`).on('shown.bs.modal', (e) => {
+      // ledger listeners
+      if (IPC_RENDERER) {
+        IPC_RENDERER.on("ledger:tokenSignature", handleSignature);
+        IPC_RENDERER.on("ledger:tokenSignatureValid", handleVerify);
+      }
+      // verify token information when opening modal
+      verifyToken();
+    });
+
+    return () => {
+      // remove all listeners on unmount
+      $(`#${modalId}`).off();
+    }
+  }, []);
+
+  // token change -> go to begin and verify it
+  useEffect(() => {
+    $(`#${modalId}`).modal('hide');
+  }, [token]);
+
+  /**
+   * Render
+   */
+
+  const renderOkMessage = () => {
+    return (
+      <div>
+        <label>{t`Token was signed!`}</label>
+      </div>
+    );
+  }
+
+  const renderClose = () => {
+    return (
+      <button onClick={closeModal} type="button" className="btn btn-secondary">{t`Close`}</button>
+    );
+  }
+
+  const renderSendButton = () => {
+    return (
+        <button type="button" className="btn btn-hathor" onClick={handleSend}>{t`Sign token`}</button>
+    );
+  }
+
+  const renderFooter = () => {
+    // Don't render the footer separator if there is no footer
+    // When waitingLedger is true, showSend is always false
+    if (waitingLedger) return null;
```

Comment:
Maybe here we should return a 'Waiting for ledger...' or something like that

Reply:
Done.